dataland-dummy-qa-service/src/main/kotlin/org/dataland/datalanddummyqaservice/services/QaService.kt

```diff
@@ -0,0 +1,48 @@
+package org.dataland.datalanddummyqaservice.services
+
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.dataland.datalandmessagequeueutils.exceptions.MessageQueueException
+import org.slf4j.LoggerFactory
+import org.springframework.amqp.core.Message
+import org.springframework.amqp.rabbit.annotation.RabbitListener
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.ComponentScan
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a QA Service reacting on the upload_queue and forwarding message to qa_queue
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
+ */
+@Component
+@ComponentScan("org.dataland")
+class QaService(
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Method to retrieve message from upload_queue and constructing new one for qa_queue
+     * @param message Message retrieved from upload_queue
+     */
+    @RabbitListener(queues = ["upload_queue"])
+    fun receive(message: Message) {
+        val dataId = cloudEventMessageHandler.bodyToString(message)
+        val correlationId = message.messageProperties.headers["cloudEvents:id"].toString()
+        if (!dataId.isNullOrEmpty()) {
+            logger.info(
+                "Received data upload with DataId: $dataId on QA message queue with Correlation Id: " +
+                    correlationId,
+            )
+            cloudEventMessageHandler.buildCEMessageAndSendToQueue(
+                dataId, "QA Process Completed", correlationId,
+                "qa_queue",
+            )
+        } else {
+            val internalMessage = "Error receiving information for QA service. Correlation ID: $correlationId"
+            logger.error(internalMessage)
+            throw MessageQueueException(
+                "Error receiving data for QA process: $internalMessage",
```

Comment:
what happens if this error is thrown? WIll teh message go to the rejected-queue?

Reply:
Currently the error message will be only logged, as we do not have a rejected_queue or dead letter queue yet.

Reply:
ok